src/containers/Application/ApplicationPage.tsx

```diff
@@ -43,6 +47,14 @@ const ApplicationPage: React.FC<ApplicationProps> = ({ structure }) => {
     // TO-DO: Redirect based on Progress (wait till Progress calculation is done)
   }, [structure])
 
+  const handleChangeToPage = (sectionCode: string, pageNumber: number) => {
+    if (!structure.info.isLinear)
+      push(`/applicationNEW/${structure.info.serial}/${sectionCode}/Page${pageNumber}`)
```

Comment:
Would prefer for progress component to do within itself

Reply:
Done in big commit: 32bf4f0